### ISSUE TITLE
hide private classes

### ISSUE THREAD

**Issue by umlaeute** [thestk/rtmidi#136]
it seems that the backend classes are meant to be a private implementation details, and should not be exposed in `RtMidi.h`

**Comment by umlaeute** [comment]
i tried to search github repositories, and the only uses of `MidiInJack` i found so far were verbatim copies of `RtMidi.cpp`, `RtMidi.h` or its documentation.

However, after wading through 400 repositories (with 1718 hits)  I managed to hit the rate-limit of github (and trigger an abuse detection mechanism), so the search was not exhaustive yet.

I also did a (possible not exhaustive) search on https://codesearch.debian.net with similar results.
I'm pretty sure that these classes are not used directly by any consumer.
